detox/src/devices/common/drivers/android/exec/AAPT.test.js

```diff
@@ -1,70 +1,89 @@
-//Disabled until we can create a build environment for Android in CI
-xdescribe('AAPT', () => {
-  let AAPT;
-  let aapt;
-  let exec;
+const aaptMockResults = require('./__mocks__/aapt-results');
+
+describe('AAPT', () => {
+  const mockAPKPath = 'path/to/app.apk';
+  const mockAAPTPath = 'mockSdk/build-tools/30.0.0/aapt';
+  const mockEscapeFunc = (source) => `escaped(${source})`;
 
+  const givenAAPTResult = (result) => {
+    exec.mockResolvedValue({
+      stdout: result,
+    });
+  };
+  const execCommand = () => exec.mock.calls[0][0];
+
+  let environment;
+  let escape;
+  let exec;
+  let aapt;
   beforeEach(() => {
-    jest.mock('../../../../utils/logger');
-    jest.mock('child-process-promise');
-    exec = require('child-process-promise').exec;
+    jest.mock('../../../../../utils/environment');
+    environment = require('../../../../../utils/environment');
+    environment.getAaptPath.mockResolvedValue(mockAAPTPath);
+
+    jest.mock('../../../../../utils/exec');
+    exec = require('../../../../../utils/exec').execWithRetriesAndLogs;
+    givenAAPTResult('');
+
+    jest.mock('../../../../../utils/pipeCommands');
+    escape = require('../../../../../utils/pipeCommands').escape.inQuotedString;
+    escape.mockImplementation(mockEscapeFunc);
 
-    AAPT = require('./AAPT');
+    const AAPT = require('./AAPT');
```

Comment:
I think this should be extracted from the `beforeEach`.

Reply:
Discussed f2f